spacy/tests/lang/hi/test_lex_attrs.py

```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import pytest
+from spacy.lang.hi.lex_attrs import norm, like_num
+
+
+def test_hi_tokenizer_handles_long_text(hi_tokenizer):
+    text = """
+ये कहानी 1900 के दशक की है। कौशल्या (स्मिता जयकर) को पता चलता है कि उसका 
+छोटा बेटा, देवदास (शाहरुख खान) वापस घर आ रहा है। देवदास 10 साल पहले कानून की 
+पढ़ाई करने के लिए इंग्लैंड गया था। उसके लौटने की खुशी में ये बात कौशल्या अपनी पड़ोस 
+में रहने वाली सुमित्रा (किरण खेर) को भी बता देती है। इस खबर से वो भी खुश हो जाती है।
+"""
+    tokens = hi_tokenizer(text)
+    assert len(tokens) == 86
+
+
+@pytest.mark.parametrize(
+    "word,word_norm",
+    [
+        ("चलता", "चल"), 
+        ("पढ़ाई", "पढ़"), 
+        ("देती", "दे"), 
+        ("जाती", "ज"),
```

Comment:
("जाती", "जा")

Reply:
I am not sure what is a `word_norm`, but ज is just a letter and not a dictionary word

Reply:
Or did you mean the technical root of the words in the same 'family'?

Reply:
Good point, Akshay. 

> but ज is just a letter and not a dictionary word 

You are right 

My understanding is that normalization is more similar to stemming than lemmatization. The output of lemmatizer has to be a dictionary word, while the stemmer tries to reduce words to a canonical word which may not be present in the dictionary.  

For example, in this article 
https://research.variancia.com/hindi_stemmer/
"ख़रीदारों के लिए मार्ग दर्शिका"  is normalized to खरीदार के लिए मार्ग दर्शिक
where दर्शिक is not a dictionary word 

Check out the English stemmer http://snowball.tartarus.org/algorithms/english/stemmer.html reduces these words 
conspirator
conspirators
conspire
conspired
conspiring

to conspir which is not a dictionary word.

Reply:
Got it. That makes sense.